purchases/src/main/kotlin/com/revenuecat/purchases/common/utils.kt

```diff
@@ -86,3 +87,17 @@ internal fun String.sha256() =
 
 internal val Context.versionName: String?
     get() = this.packageManager.getPackageInfo(this.packageName, 0).versionName
+
+val Context.playStorePackageInfo: PackageInfo?
+    get() = try {
+        this.packageManager.getPackageInfo("com.android.vending", 0)
+    } catch (e: Throwable) {
+        null
+    }
+
+val Context.playServicesPackageInfo: PackageInfo?
+    get() = try {
+        this.packageManager.getPackageInfo("com.google.android.gms", 0)
```

Comment:
I have not found official documentation about this but in my tests, pre-installed apps did not require the QUERY_ALL_PACKAGES permission to get information about them so this should be fine.

Reply:
We should confirm this in different android versions, including the upcoming android 14. But yeah, maybe getting package info of a specific package works...

Reply:
Will test on the latest version, and try to find a more definitive source of this permission exception

Reply:
Found the answer: "Some system packages that implement core Android functionality are automatically visible to your app, even when your app targets Android 11 or higher."

https://developer.android.com/training/package-visibility/automatic#system-packages-visible-automatically